src/types-and-traits.rst

```diff
@@ -2101,31 +2101,54 @@ Type Inference
 :t:`[expression]s` and :t:`[pattern]s` within a :t:`type inference root`.
 
 :dp:`fls_ybvrhh96fc7y`
-A :t:`type inference root` is an :t:`expression` whose inner :t:`[expression]s`
+A :t:`type inference root` is a :t:`construct` whose inner :t:`[expression]s`
 and :t:`[pattern]s` are subject to :t:`type inference` independently of those
 found in other :t:`[type inference root]s`.
 
 :dp:`fls_EWBilpepaDcX`
-The following :t:`[expression]s` are considered :t:`[type inference root]s`:
+The following :t:`[construct]s` are considered :t:`[type inference root]s`:
 
 * :dp:`fls_NYSzcvf5nQpi`
   A :t:`constant argument`.
 
 * :dp:`fls_htLp5J5ObgNh`
   A :t:`constant initializer`.
 
-* :dp:`fls_cPlCLGCcl7EK`
-  A :t:`static initializer`.
-
 * :dp:`fls_KphY5qHev0Dc`
   The :t:`discriminant initializer` of an :t:`enum variant`.
 
 * :dp:`fls_Sowatt1V988J`
   A :t:`function body`.
 
+* :dp:`fls_dF4kA29a4iMF`
+  A :t:`let initializer`.
```

Comment:
`let` is not a type inference root, it is inferred together with all other statements in the containing root

Reply:
I don't think so. A `let statement` has an optional `type ascription`. If the ascription is present, then the ascription imposes its type on the `let initializer` (more precisely the expression of the let initializer). Otherwise how do we trigger type inference for `let statement`s?

Reply:
It's inferred as part of the containing inference root. Inference collects all type ascriptions and known types (including those in path expressions like `Option::<u8>::None`), creates inference variables for everything that isn't known yet, and then propagates the known types through inference variables via unification (and trait solving and method resolution etc.).

Ascriptions can also be incomplete, like in `let x: Option<_>`, and in that case inference has to figure out what the `_` stands for using the surrounding context (not just the initializer). Every inference root is inferred in isolation, so this would not be possible if `let` was one.

Reply:
@Veykril Lukas, what do you think about the wording, and Jonas's points?

Reply:
After discussing this with Lukas live, I understand your point Jonas. Fixed.

Reply:
Hmm, is the diff wrong or are they still listed as inference roots?